### PR TITLE
RHCLOUD-29443 | Remove the secrets' migration code and the feature flag

### PR DESCRIPTION
**Pull request by MikelAlejoBR**
After migrating all the secrets to sources, we decided that it was time
to remove both the database columns for the secrets and the feature
flag, since storing secrets in Sources is going to be enabled by
default.

## Dependencies
* https://github.com/RedHatInsights/notifications-backend/pull/2348

## Jira ticket
[[RHCLOUD-29443]](https://issues.redhat.com/browse/RHCLOUD-29443)

### CODE REVIEW

File: database/src/main/resources/db/migration/V1.89.0__RHCLOUD-29443_remove_secrets_columns.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+ALTER TABLE
```
Comment (g-duval) [inline database/src/main/resources/db/migration/V1.89.0__RHCLOUD-29443_remove_secrets_columns.sql:1]: I think we should move this Drop column script to another PR, and merge it a little bit after, just in case we have to rollback the prod delivery for any reason.

Comment (MikelAlejoBR) [reply database/src/main/resources/db/migration/V1.89.0__RHCLOUD-29443_remove_secrets_columns.sql]: You are right. I have removed the migration but we have to coordinate to merge this one, because I need to run the migrations first before removing all the code!

**Comment by gwenneg** [comment]
Where are we on this topic?

- Are we done with the migration of the DB data in all environments?
- Does this PR needs a rebase/update or can it be reviewed/merged as is?

**Comment by MikelAlejoBR** [comment]
> Where are we on this topic?
> 
>     * Are we done with the migration of the DB data in all environments?

Not yet. My idea was to coordinate the data migration with the merge of this PR, and then do a release to stage/prod. So I would love to do that early on the week. 

>     * Does this PR needs a rebase/update or can it be reviewed/merged as is?

I think it can be reviewed because I did put the schema migration script on a different PR.

**Comment by gwenneg** [comment]
Perfect, let's do that next week! Thanks for the update.

File: engine/src/test/java/com/redhat/cloud/notifications/events/LifecycleITest.java
Comment (gwenneg) [inline engine/src/test/java/com/redhat/cloud/notifications/events/LifecycleITest.java:1]: Sources is no longer available in the engine. All changes made in this class need to be reverted.

File: backend/src/test/java/com/redhat/cloud/notifications/routers/EndpointResourceTest.java [hunk 2]
```diff
@@ -1599,6 +1719,23 @@ void testConnectionCount() {
             ep.setEnabled(true);
             ep.setProperties(properties);
 
+            // Mock the Sources service calls.
+            final Secret secretTokenSecret = new Secret();
+            secretTokenSecret.id = new Random().nextLong(1, Long.MAX_VALUE);
+            secretTokenSecret.password = properties.getSecretToken();
```
Comment (gwenneg) [inline backend/src/test/java/com/redhat/cloud/notifications/routers/EndpointResourceTest.java:1725]: This and other changes from this PR are repeated several times. Could you please lower the amount of duplicated code?

**Review by gwenneg — APPROVED** [review]
LGTM

File: backend/src/test/java/com/redhat/cloud/notifications/routers/EndpointResourceTest.java (hunk 2 above)
Comment (MikelAlejoBR) [reply backend/src/test/java/com/redhat/cloud/notifications/routers/EndpointResourceTest.java]: Thank you for taking care of this. I was finishing some work for the service accounts that got in the way and was planning to review this today.
